src/shogun/lib/sg_types.h

```diff
@@ -0,0 +1,126 @@
+#include <shogun/lib/any.h>
+#include <type_traits>
```

Comment:
should also move the base types here,
https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/base/base_types.h

Reply:
shouldn't that go to a sg_type_traits.h file?